### PR TITLE
For Jira - CAMEL-15257 - Component for Jsonata Integration

### PR DESCRIPTION
**Pull request by jamesmacinnes**
Initial implmentation of: camel jsonata integration

[ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
[ ] Each commit in the pull request should have a meaningful subject line and body.
[ ] If you're unsure, you can format the pull request title like `[CAMEL-XXX] Fixes bug in camel-file component`, where you replace `CAMEL-XXX` with the appropriate JIRA issue.
[ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
[ ] Run `mvn clean install -Psourcecheck` in your module with source check enabled to make sure basic checks pass and there are no checkstyle violations. A more thorough check will be performed on your pull request automatically.
Below are the contribution guidelines:
https://github.com/apache/camel/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: components/camel-jsonata/pom.xml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        <dependency>
+            <groupId>com.fasterxml.jackson.core</groupId>
+            <artifactId>jackson-databind</artifactId>
+        </dependency>
+
+        <!-- https://mvnrepository.com/artifact/com.ibm.jsonata4java/JSONata4Java -->
+        <dependency>
+            <groupId>com.ibm.jsonata4java</groupId>
+            <artifactId>JSONata4Java</artifactId>
+            <version>1.0.0</version>
```
Comment (oscerd) [inline components/camel-jsonata/pom.xml:57]: This should go as property placeholder at parent pom

File: components/camel-jsonata/src/generated/resources/org/apache/camel/component/jsonata/jsonata.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
+{
+  "component": {
+    "kind": "component",
+    "name": "jsonata",
+    "title": "JSONATA",
+    "description": "JSON to JSON transformation using JSONATA.",
+    "deprecated": false,
+    "firstVersion": "3.4.0",
```
Comment (oscerd) [inline components/camel-jsonata/src/generated/resources/org/apache/camel/component/jsonata/jsonata.json:8]: You need to regenerate the src/generated.. it's 3.5.0

File: components/camel-jsonata/src/main/java/org/apache/camel/component/jsonata/JsonataEndpoint.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        
+        JsonNode input;
+        ObjectMapper mapper = new ObjectMapper();
+        if (getInputType() == JsonataInputOutputType.JsonString) {
+            input =  mapper.readTree(exchange.getIn().getBody(InputStream.class));
+        } else {
+            input = (JsonNode)exchange.getIn().getBody();
+        }
+
+        JsonNode output=null;
```
Comment (oscerd) [inline components/camel-jsonata/src/main/java/org/apache/camel/component/jsonata/JsonataEndpoint.java:107]: Please run man clean install -Psourcecheck. There are code style errors

File: components/camel-jsonata/src/test/java/org/apache/camel/component/jolt/JsonataFirstSampleTest.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+import org.junit.jupiter.api.Test;
+
+/**
+ * Unit test based on the first sample test from the Jsonata project.
+ */
+public class JsonataFirstSampleTest extends CamelTestSupport {
+
+    @Test
+    public void testFirstSampleJsonata() throws Exception {
+        //getMockEndpoint("mock:result").expectedMinimumMessageCount(1);
```
Comment (oscerd) [inline components/camel-jsonata/src/test/java/org/apache/camel/component/jolt/JsonataFirstSampleTest.java:37]: Please remove

File: components/camel-jsonata/pom.xml (hunk 1 above)
Comment (jamesmacinnes) [reply components/camel-jsonata/pom.xml]: Moved to property in parent pom.

File: components/camel-jsonata/src/generated/resources/org/apache/camel/component/jsonata/jsonata.json (hunk 2 above)
Comment (jamesmacinnes) [reply components/camel-jsonata/src/generated/resources/org/apache/camel/component/jsonata/jsonata.json]: Done.

File: components/camel-jsonata/src/main/java/org/apache/camel/component/jsonata/JsonataEndpoint.java (hunk 3 above)
Comment (jamesmacinnes) [reply components/camel-jsonata/src/main/java/org/apache/camel/component/jsonata/JsonataEndpoint.java]: Done.

File: components/camel-jsonata/src/test/java/org/apache/camel/component/jolt/JsonataFirstSampleTest.java (hunk 4 above)
Comment (jamesmacinnes) [reply components/camel-jsonata/src/test/java/org/apache/camel/component/jolt/JsonataFirstSampleTest.java]: Done.

**Comment by jamesmacinnes** [comment]
Ah, I didn't notice there was that newer version. Let me try it out and repush.

**Comment by oscerd** [comment]
Looks good to me, I'll merge the PR this week. Thank you @jamesmacinnes

**Comment by jamesmacinnes** [comment]
@oscerd did a repush as I got a build error on the autobuild on docker component test code. Did a fresh full build on local and didn't get that error. Did a recommit and push to see if that docker error happens again.

**Comment by oscerd** [comment]
> @oscerd did a repush as I got a build error on the autobuild on docker component test code. Did a fresh full build on local and didn't get that error. Did a recommit and push to see if that docker error happens again.

Yeah, I fixed the codestyle issue an hour ago, it's not your fault. It should be ok now :-)

**Comment by oscerd** [comment]
Thanks, it has been merged and will be released in 3.5.0

**Pull request closed without merging**